src/Html/HtmlBuilder.php

```diff
@@ -474,6 +474,20 @@ public static function limit($html, $maxLength = 100, $end = '...')
         return $result;
     }
 
+    /**
+     * Gets the readable length of a string for limit calculations.
+     *
+     * Considers multiple spaces and line breaks to be 1 character, regardless of OS.
+     *
+     * @param string $str
+     * @return int
+     */
+    protected static function getReadableLength($str)
+    {
+        $str = str_replace(["\n", "\r"], "\n", $str);
+        return strlen(preg_replace('/\s+/', ' ', $str));
```

Comment:
@bennothommo could you expand on why this is necessary? Also, doesn't this turn Windows line breaks into two unix linebreaks?

Reply:
@LukeTowers as far as I can tell from the intent of the `HtmlBuilder::limit()` method, it's trying to limit the characters of a HTML string whilst allowing tags to be correctly used in the limited string.

I made this change to calculate the "readable" length of the string when determining where to crop the string. Take the following unit test:
```
<p>The quick brown fox jumped          over the lazy dog</p>\r\n<p>The quick brown fox jumped over the lazy dog</p>
```
This previously would've resulted in the string being trimmed a lot further, as each space character was considered to be one character, even though when it's actually rendered in the browser, it's just one space character as browsers ignore multiple whitespaces. Simultaneously with the CRLF line break (`\r\n`) being two characters, when it's really just one "character".

My change effectively makes it so that both these instances are considered as one character, effectively including more "readable" characters in the limit.

As for your second point, it seems to be passing the tests, but you're right - I will adjust that.

Reply:
I've removed the line break change line, as the next line handles it anyway (preg `\s` matches newlines too)